pyk/src/pyk/cterm/symbolic.py

```diff
@@ -94,6 +96,28 @@ def kast_to_kore(self, kinner: KInner) -> Pattern:
     def kore_to_kast(self, pattern: Pattern) -> KInner:
         return kore_to_kast(self._definition, pattern)
 
+    def minimize_constraints(self, constraints: tuple[KInner, ...], pc: KInner) -> tuple[KInner, ...]:
+        """Minimize given branching constraints with respect to a given path condition."""
+        # Determine intersection between all returned sets of branching constraints
+        flattened_default = tuple(flatten_label('#And', c) for c in constraints)
+        intersection = flattened_default[0]
+        for i in range(1, len(flattened_default)):
+            intersection = [c for c in intersection if c in flattened_default[i]]
+            # If intersection is empty, there is nothing to be done
+            if not intersection:
+                return constraints
```

Comment:
Consider using `set.intersection`:

```suggestion
        _intersection = set.intersection(*(set(cs) for cs in flattened_default))
        intersection = tuple(c for c in flattened_default[0] if c in _intersection)
```

Both this and the original implementation will fail on empty `constraints` though, so additionally that case has to be handled.

Reply:
By construction, this function should not be called with empty `constraints` or just one set of constraints. I'll add an appropriate assertion.